24.0.0.1 staging (#3520)
OpenLiberty/blogs: posts/2024-01-30-24.0.0.1.adoc

---
layout: post

> title: "Expanded InstantOn support in 24.0.0.1"

Reviewer: ```suggestion
title: "Expanded InstantOn support in Open Liberty 24.0.0.1"
```

# Do NOT change the categories section
categories: blog
author_picture: https://avatars3.githubusercontent.com/dmuelle
author_github: https://github.com/dmuelle

> seo-title: Expanded InstantOn support in 24.0.0.1 - OpenLiberty.io

Reviewer: ```suggestion
seo-title: Expanded InstantOn support in Open Liberty 24.0.0.1 - OpenLiberty.io
```

seo-description: InstantOn now supports features that enable Jakarta and Java XML Web Services and Jakarta and Java Mail functions, as well as more versions of previously supported features. This release also introduces the option to verify the authenticity of the Open Liberty public key.
blog_description: InstantOn now supports features that enable Jakarta and Java XML Web Services and Jakarta and Java Mail functions, as well as more versions of previously supported features. This release also introduces the option to verify the authenticity of the Open Liberty public key.
open-graph-image: https://openliberty.io/img/twitter_card.jpg
open-graph-image-alt: Open Liberty Logo
---

> = Expanded InstantOn support in 24.0.0.1

Reviewer: ```suggestion
= Expanded InstantOn support in Open Liberty 24.0.0.1
```

David Mueller <https://github.com/dmuelle>
:imagesdir: /
:url-prefix:
:url-about: /
//Blank line here is necessary before starting the body of the post.

The 24.0.0.1 release expands the set of Liberty features that Liberty InstantOn supports. InstantOn now supports features that enable Jakarta and Java XML Web Services and Jakarta and Java Mail functions, and more versions of previously supported features. This release also introduces the option to verify the authenticity of the Open Liberty public key.

In link:{url-about}[Open Liberty] 24.0.0.1:

> * <<instanton_features, Instant On support for jaxws, xmlws, and other features>>

Reviewer: ```suggestion
* <<instanton_features, InstantOn support for Jakarta XML Web Services, Java Web Services
 and other features>>
```

[source,gradle]
----
buildscript {
    repositories {
        mavenCentral()
    }
    dependencies {
        classpath 'io.openliberty.tools:liberty-gradle-plugin:3.8'
    }
}
apply plugin: 'liberty'
----

Or if you're using link:{url-prefix}/docs/latest/container-images.html[container images]:

[source]
----
FROM icr.io/appcafe/open-liberty
----

Or take a look at our link:{url-prefix}/start/[Downloads page].

If you're using link:https://plugins.jetbrains.com/plugin/14856-liberty-tools[IntelliJ IDEA], link:https://marketplace.visualstudio.com/items?itemName=Open-Liberty.liberty-dev-vscode-ext[Visual Studio Code] or link:https://marketplace.eclipse.org/content/liberty-tools[Eclipse IDE], you can also take advantage of our open source link:https://openliberty.io/docs/latest/develop-liberty-tools.html[Liberty developer tools] to enable effective development, testing, debugging, and application management all from within your IDE.

// // // // DO NOT MODIFY THIS COMMENT BLOCK <GHA-BLOG-TOPIC> // // // //
// Blog issue: https://github.com/OpenLiberty/open-liberty/issues/27303
// Contact/Reviewer: anjumfatima90
// // // // // // // //
[#instanton_features]

> == Instant On support for jaxws/xmlws and other features

Reviewer: ```suggestion
== InstantOn support for Jakarta XML Web Services, Java Web Services
 and other features
```

> link:https://openliberty.io/docs/latest/instanton.html[Open Liberty InstantOn] provides fast startup times for MicroProfile and Jakarta EE applications.

Reviewer: ```suggestion
Open Liberty InstantOn provides fast startup times for MicroProfile and Jakarta EE applications.
```

> InstantOn supports link:https://openliberty.io/docs/latest/instanton.html#supported-features[a subset of Open Liberty features]. Any public features that are enabled outside of this set of supported features cause the checkpoint to fail with an error message. As of the 24.0.0.1 release, the following features are enhanced to support InstantOn, in addition to the set of features included initial Liberty InstantOn 23.0.0.6 release:

Reviewer: ```suggestion
InstantOn supports link:{url-prefix}/docs/latest/instanton.html#supported-features[a subset of Open Liberty features]. As of the 24.0.0.1 release, the following features are enhanced to support InstantOn, in addition to the set of features included in the link:{url-prefix}/blog/2023/06/27/23.0.0.6.html[initial Liberty InstantOn release]:
```

- `jaxws-2.2` (24.0.0.1 and later)
- `xmlWS-3.0` (24.0.0.1 and later)
- `xmlWS-4.0` (24.0.0.1 and later)
- `jdbc-4.1` (23.0.0.9 and later)
- `jdbc-4.3` (23.0.0.9 and later)
- `appSecurity-1.0` (23.0.0.10 and later)
- `javaMail-1.6` (23.0.0.10 and later)
- `mail-2.0` (23.0.0.10 and later)
- `mail-2.1` (23.0.0.10 and later)
- `passwordUtilities-1.0` (23.0.0.10 and later)

> - `localConnector-1.0` (23.0.0.11 and later)

Reviewer: I'm fine with keeping this as-is, but we can also consider using the full names for the features, for example:
- Java Web Services 2.2 (`jaxws-2.2`) - 24.0.0.1 and later
- Jakarta XML Web Services (`xmlWS-3.0`) -  24.0.0.1 and later
.....

Another option would be to make this into a 3 column table: Name, Feature, Initial InstantOn support

Reviewer 2: added the table and security tag
